LogfileQueueReader/LogfileQueueReaderExample.cpp

```diff
@@ -0,0 +1,271 @@
+ *
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ *
+ * Regarding setting node name using Logfile Reader and how that relates
+ * to setFilePath(), setSessionId(), and setNodeName():
+ *
+ * Example: in directory 1234, let a logfile be name.5678.plog,
+ * where 1234 = session ID, 5678 = GUID, and name = node name.
+
+ * To read from this logfile, you can use its filepath or its session ID.
+ *
+ * Session ID logic is default write logic.
+ * If you do not set explicit filepath, then directory will be timestamp
+ * of write time, and name will be whatever is set in setNodeName().
+ *
+ * setFilePath() logic overrides default Session ID logic.
+ * For Logfile Reader, using a Logfile's explicit filepath is strongly
+ * recommended.
+ *
+ * Option 1: Using explicit filepath to read logfiles (recommended):
+ * Any file specified in the path can be read from;
+ * node name is irrelevant.
+ * How:
+ * 1. use any setNodeName(), then
+ * 2. use setFilePath() in prepareLogfileToRead().
+ *
+ * Option 2: Using Session ID to read logfiles:
+ * Since reader's SDF key must match written SDF key, it's required to
+ * use the same node reference/context to write to, and then read from,
+ * a Logfile using its SessionID. Generally, that means writing to a
+ * Logfile and then reading from that Logfile in the same "example"
+ * file (same Node called from main() with connectPolySync() ).
+ * Doing this guarantees the same node reference, and hence, the same SDF Key.
+ * How:
+ * Not covered in this example.
+ * All C/C++ Logfile examples use explicit filepath, not Session ID.
+ */
+
+/**
+ * \example LogfileQueueReaderExample.cpp
+ *
+ * PolySync Logfile CPP API Queue Reader Example.
+ *
+ * Shows how to use the Logfile API to read logfiles from the async queue.
+ *
+ * @file LogfileQueueReaderExample.cpp
+ * @brief LogfileReader Source
+ *
+ */
+
+
+#include "LogfileTestNode.hpp"          // For all CPP Logfile API examples.
+
+using namespace std;
+
+using namespace polysync::datamodel;
+
+
+void LogFileTestNode::prepareLogfileToRead()
+{
+    // 1. filter out certain message types from being published (optional).
+    //    filterOutMessages();
+
+    // 2. Choose Read method: either filepath (simplest use case), or session ID.
+    // Option 1: set filepath. If set by writer in same file, no need to reset.
+    _logFile->setFilePath( "/tmp/polysync_logfile.plog"  );
+
+    _logFile->setSessionId( 1 );
+
+    // Option 2: Using setSessionId() instead of setFilePath():
+    //_logFile->setSessionId( 1234 );
```

Comment:
That does seem unclear doesn't it? Improved the documentation https://github.com/PolySync/PolySync-CPP-Examples/pull/13/commits/4459cbb9d677792ef6e926bdda92d006acffea2b